### PR TITLE
fix: Python installation when running outside skbuild

### LINKED ISSUE #330: Manual python installation does not install to package folder
Observed in #327. It should install to `spglib` package in the install dir.

@lan496 @atztogo  Quick question about #317. Is a bug-fix and not a general convention change?

### LINKED ISSUE #344: CMake fails to find Python library for tests
I am building spglib v2.1.0 with CMake on Debian unstable:
```bash
mkdir build && cd build
cmake .. -DSPGLIB_WITH_Fortran=ON -DSPGLIB_WITH_Python=ON
make
make test
```
Tests fail for python-tests with `ModuleNotFoundError: No module named 'spglib'`. I tried setting `PYTHONPATH` to various locations seemingly containing Python files after the build, but I did not succeed.

### LINKED ISSUE #429: Python interface is broken: ModuleNotFoundError: No module named 'spglib._version'
```import spglib``` fails:
```
$ python3.9
Python 3.9.18 (main, Dec 10 2023, 01:22:40) 
[Clang 16.0.6 (https://github.com/llvm/llvm-project.git llvmorg-16.0.6-0-g7cbf1 on freebsd14
Type "help", "copyright", "credits" or "license" for more information.
>>> import spglib
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.9/site-packages/spglib/__init__.py", line 37, in <module>
    from ._version import __version__, __version_tuple__  # noqa: F401
ModuleNotFoundError: No module named 'spglib._version'
```

This is perhaps due to an older py39-setuptools_scm7-7.1.0, which isn't at the latest version 8.x

This line in ```__init__.py``` is a culprit:
> from ._version import __version__, __version_tuple__  # noqa: F401

Please fix this because many distros have setuptools_scm at 7.x or older.

### PR DESCRIPTION
**Pull request by LecrisUT**
- Fixed the python installation
- Refactored the test and execution so that installation is not required for doing pytest

Closes: #330 #344 #429
Depends on: #397

### CODE REVIEW

**Comment by LecrisUT** [comment]
Welp, 8d1ff02e7dcf8aa1f4832330865f375a8080529a has found a bunch of address sanitization issues :skull:. It is also rather tricky to get the address sanitization working on the pytest themselves. I think #303 really needs to be implemented and move all of the tests into ctest. That way we can also minimize the number of builders.

**Comment by lan496** [comment]
I do not fully understand the context, but do we need address sanitizer for the Python interface? 
It seems somewhat overwhelming for Spglib's Python developers to move Python tests into ctest.

**Comment by LecrisUT** [comment]
The core issue is not with the python interface having the address sanitizer, but that the C library tests do not cover all cases. That's why the main thing to do is #303 so we can move the main tests to ctest and not have the pytests run with sanitizers (at least not always)

**Comment by lan496** [comment]
@LecrisUT Please assign me when ready for review.

**Comment by LecrisUT** [comment]
All 3 PRs are ready, but first we should wait on all of the packagings to catch up to find if everything is ok. We can track it better after #408.

After that we should review them in order, otherwise it is a bit much to navigate:
#397 -> #335 -> #308

**Comment by lan496** [comment]
Let me review this PR next.

File: .github/workflows/step_test.yaml [hunk 1]
```diff
@@ -73,12 +73,11 @@ jobs:
...
-        # TODO: Replace with editable install once configure presets are available
-      - name: Install python bindings
-        run: >
-          python3 -m pip install
-          ${{ matrix.pre && '--pre' }}
-          .[test]
+      - name: Install dependencies
+        run: |
+          python3 -m pip install pip-tools
+          pip-compile --extra=test --strip-extras pyproject.toml ${{ matrix.pre && '--pre' }}
```
Comment (lan496) [inline .github/workflows/step_test.yaml:79]: Why does this PR generate requirements.txt here? To lock package versions?

**Review by lan496 — APPROVED** [review]

File: .github/workflows/step_test.yaml (hunk 1 above)
Comment (LecrisUT) [reply .github/workflows/step_test.yaml:79]: `pip` does not have a mode to install only dependencies, and that is the closest approach I've found to do that. What I wanted to be covered there is the case where the Spglib is built without `scikit-build-core` being involved. Weird though that it didn't error out in the previous iterations where I didn't include `configure_file(_version.py.in spglib/_version.py)`

Comment (lan496) [reply .github/workflows/step_test.yaml:79]: Although I feel tricky, it makes sense. Thanks!

Comment (LecrisUT) [reply .github/workflows/step_test.yaml:79]: I fully agree. Eventually I hope the [`skbuild-cli`](https://github.com/scikit-build/scikit-build-cli) will be implemented, at which point we wouldn't need to care about such difference.

**Pull request merged**